astro/src/content/docs/operate/deploy/terraform.mdx

```diff
@@ -0,0 +1,432 @@
+
+<RemoteCode lang="hcl" title="Example Variables File" url="https://raw.githubusercontent.com/fusionauth/fusionauth-example-terraform/main/examples/import/variables.tf" />
+
+<Aside type="caution">
+For the `fusionauth_api_key` variable, the configuration option `sensitive = true` is set. This will prevent sensitive data being added to the Terraform state.
+</Aside>
+
+For the variable definitions, create a `terraform.tfvars` file with the following content and changes:
+
+* Set the `fusionauth_api_key` variable to the value of the previously created API key.
+* Set the `fusionauth_host` variable to the hostname of the FusionAuth instance you want to manage. Make sure this value is the full URL including the protocol of your FusionAuth instance. When using docker, this will typically be `http://localhost:9011`.
+* The `fusionauth_default_tenant_id` variable can be omitted for now, but you'll update it later.
+
+<RemoteCode lang="hcl" title="Example Variable Definitions" url="https://raw.githubusercontent.com/fusionauth/fusionauth-example-terraform/main/examples/import/terraform.tfvars" />
+
+The variable definition with `terraform.tfvars` is only one option for setting these variables. Review the [Assigning Values to Root Module Variables](https://developer.hashicorp.com/terraform/language/values/variables#assigning-values-to-root-module-variables) documentation for more options.
+
+Next, add the [FusionAuth Terraform Provider](https://registry.terraform.io/providers/fusionauth/fusionauth/latest) configuration in the directory by:
+
+* Opening the [FusionAuth Terraform Provider Documentation](https://registry.terraform.io/providers/fusionauth/fusionauth/latest/docs).
+* Clicking the <strong>USE PROVIDER</strong> button.
+* Copying the Hashicorp Configuration Language (HCL) provided.
+* Creating a Terraform configuration file named `main.tf`.
+* Pasting the copied HCL into that file.
+* Adding the previously created variable as `var.fusionauth_api_key` to the `provider` section
+* Adding the FusionAuth instance you plan to manage with the `var.fusionauth_host` variable in the `provider` section
+
+Here's a screenshot of the page.
+
+![Downloading the HCL for the FusionAuth provider.](/img/docs/operate/deploy/terraform-provider.png)
+
+Here's what your `main.tf` file should look like at this point.
+
+<RemoteCode lang="hcl" title="Example main.tf File" url="https://raw.githubusercontent.com/fusionauth/fusionauth-example-terraform/main/examples/import/main.tf" tags="terraformProvider" />
+
+Finally, the following command prepares the current working directory for use with Terraform.
+
+```shell
+terraform init
+```
+
+It is always safe to run [`terraform init`](https://developer.hashicorp.com/terraform/cli/commands/init) multiple times, to bring the working directory up to date with changes in the configuration. Though subsequent runs may give errors, this command will never delete your existing configuration or state.
```

Comment:
Maybe add something about "If you update the provider version in the future, you will need to run `terraform init -upgrade` to install the new provider version locally"?

Reply:
It will tell you in the terraform init output to execute with upgrade, if you have upgraded the provider, and as the doc is already long I did choose to leave it out.